test/common.jl

```diff
@@ -19,3 +19,8 @@ else
     ArrayType = Array
     synchronize() = nothing
 end
+
+const MPIUnionTypes = Union{Char,
+                            Int8, Int16, Int32, Int64,
+                            UInt8, UInt16, UInt32, UInt64,
+                            Float32, Float64, ComplexF32, ComplexF64} |> Base.uniontypes
```

Comment:
Missing newline
```suggestion
                            Float32, Float64, ComplexF32, ComplexF64} |> Base.uniontypes

```
Maybe also add a comment to explain these are the types we want to test with MPI and warn to not remove any of them? :smile:

Reply:
Done. Also, I don't think we need `Base.uniontypes` right ?
IIUC, listing the types in a vector should be sufficient.

Reply:
Yeah, can also just be a vector.  In #689 suggested this is just a leftover from the past, so we can change it for whatever we need in the tests, which is always an iterable from what I can tell.

Reply:
It might be worth keeping some type alias for native MPI types (which would exclude Char)

Reply:
Sorry I don't get this could you clarify ?